python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -0,0 +1,265 @@
+        all the way up to nanoseconds.
+        See strftime documentation for more information on choices:
+        https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior.
+    unit : str, default 'ns'
+        The unit of the arg (D,s,ms,us,ns) denote the unit, which is an
+        integer or float number. This will be based off the
+        origin(unix epoch start).
+        Example, with unit='ms' and origin='unix' (the default), this
+        would calculate the number of milliseconds to the unix epoch start.
+    infer_datetime_format : bool, default False
+        If True and no `format` is given, attempt to infer the format of the
+        datetime strings, and if it can be inferred, switch to a faster
+        method of parsing them. In some cases this can increase the parsing
+        speed by ~5-10x.
+
+    Returns
+    -------
+    datetime
+        If parsing succeeded.
+        Return type depends on input:
+        - list-like: DatetimeIndex
+        - Series: Series of datetime64 dtype
+        - scalar: Timestamp
+
+    Examples
+    --------
+    Assembling a datetime from multiple columns of a DataFrame. The keys can be
+    common abbreviations like ['year', 'month', 'day', 'minute', 'second',
+    'ms', 'us', 'ns']) or plurals of the same
+    >>> df = cudf.DataFrame({'year': [2015, 2016],
+    ...                    'month': [2, 3],
+    ...                    'day': [4, 5]})
+    >>> cudf.to_datetime(df)
+    0   2015-02-04
+    1   2016-03-05
+    dtype: datetime64[ns]
+    >>> cudf.to_datetime(1490195805, unit='s')
+    numpy.datetime64('2017-03-22T15:16:45.000000000')
+    >>> cudf.to_datetime(1490195805433502912, unit='ns')
+    numpy.datetime64('1780-11-20T01:02:30.494253056')
+    """
+    if arg is None:
+        return None
+
+    if exact is False:
+        raise NotImplementedError("exact support is not yet implemented")
+
+    if origin != "unix":
+        raise NotImplementedError("origin support is not yet implemented")
+
+    if yearfirst:
+        raise NotImplementedError("yearfirst support is not yet implemented")
+
+    try:
+        if isinstance(arg, cudf.DataFrame):
+            # we require at least Ymd
+            required = ["year", "month", "day"]
+            req = sorted(list(set(required) - set(arg._data.names)))
+            if len(req):
+                raise ValueError(
+                    "to assemble mappings requires at least that "
+                    "[year, month, day] be specified: [{required}] "
+                    "is missing".format(required=",".join(req))
+                )
+
+                # replace passed unit with _unit_map
+
+            def get_units(value):
+                if value in _unit_map:
+                    return _unit_map[value]
+
+                # m is case significant
+                if value.lower() in _unit_map:
+                    return _unit_map[value.lower()]
+
+                return value
```

Comment:
Can roll this function outside this definition so it's not redefined at every call.